### PR TITLE
Fix CI fail when sending GH comment

### LINKED ISSUE #546: Failure to send a comment fails the build
As in a build for  #542 

would it be okay to allow failures in this section of the build ? what do you think ?

### PR DESCRIPTION
**Pull request by PaNaVTEC**
This fixes #546 

What was happening is that we were opening pull request while the CI was building the last commit  that we pushed.

I modified the script to:
- Don't fail the build if the github comment can't be sent.
- If the CI is building and we don't have a PR but at the time that we are about to send the comment we have created the PR, the modification on the script will get the PR number and the comment will be send. This is done triggering a github api call with the branch name if we don't have the PR number when we are about to send the comment.
- Check arguments and set names like `-o`. I've used a generator that you can find here: http://getoptgenerator.dafuer.es/

### CODE REVIEW

File: _utility/send_gh_comment_validation.sh [hunk 1]
```diff
@@ -0,0 +1,105 @@
+# Define help function
```
Comment (cyplo) [inline _utility/send_gh_comment_validation.sh:4]: A note on how did you generate this and how a next maintainer might re-generate with minimal effort might be helpful here

File: _utility/send_gh_comment.sh [hunk 2]
```diff
@@ -1,34 +1,43 @@
...
-  echo -e "Usage: $0 REPO_OWNER REPO_NAME PR_NUMBER AUTH_TOKEN DEPLOYMENT_URL\n"
-  echo "Example: $0 codurance site 4 token https://bucket-name.amazon.com/index.html"
-  exit 1
+REPO_OWNER=''
+REPO_NAME=''
+PR_NUMBER=''
+GITHUB_TOKEN=''
+DEPLOYMENT_URL=''
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
```
Comment (neomatrix369) [inline _utility/send_gh_comment.sh]: would a combination of ' and " on this line work - would make it easier to read and understand whats happening

File: _utility/send_gh_comment.sh [hunk 3]
```diff
@@ -1,34 +1,43 @@
 #!/bin/bash
-set -e
+set +e
```
Comment (neomatrix369) [inline _utility/send_gh_comment.sh]: Either create a usage() that displays the usage with an example or put that text in the header of this file as comments whichever is more appropriate - will help with the learning curve.

File: _utility/send_gh_comment.sh [hunk 4]
```diff
@@ -1,34 +1,43 @@
...
+REPO_OWNER=''
+REPO_NAME=''
+PR_NUMBER=''
+GITHUB_TOKEN=''
+DEPLOYMENT_URL=''
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+source "$DIR/send_gh_comment_validation.sh"
+
+if [ -z "$PR_NUMBER" ]
```
Comment (neomatrix369) [inline _utility/send_gh_comment.sh]: Maybe extract this into a function of its own, so the main body is more readable.

File: _utility/send_gh_comment.sh [hunk 5]
```diff
@@ -1,34 +1,43 @@
...
+	fi
 fi
 
-GITHUB_USERNAME='CoduranceBot'
-REPO_OWNER=$1
-REPO_NAME=$2
-PR_NUMBER=$3
-AUTH_TOKEN=$4
-DEPLOYMENT_URL=$5
+LAST_COMMENT_ID=$(curl https://api.github.com/repos/$REPO_OWNER/$REPO_NAME/issues/$PR_NUMBER/comments?access_token=$GITHUB_TOKEN \
```
Comment (neomatrix369) [inline _utility/send_gh_comment.sh]: Just like above, can be extracted into a function

File: _utility/send_gh_comment.sh [hunk 6]
```diff
@@ -1,34 +1,43 @@
...
+if [ $? -ne 0 ]; then
+  echo "Error while parsing last commend id"
+  exit 1
+fi
 
 COMMENT="Deployed: $DEPLOYMENT_URL"
-LAST_COMMENT_ID=$(curl https://api.github.com/repos/$REPO_OWNER/$REPO_NAME/issues/$PR_NUMBER/comments?access_token=$AUTH_TOKEN \
-  -X GET | jq "[.[] | select(.user.login==\"$GITHUB_USERNAME\")][0].id")
-
 if [ "$LAST_COMMENT_ID" != "null" ]; then
```
Comment (neomatrix369) [inline _utility/send_gh_comment.sh]: Extract into function

File: _utility/send_gh_comment_validation.sh [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+function help(){
```
Comment (neomatrix369) [inline _utility/send_gh_comment_validation.sh:5]: The functions that do not belong directly to this file can be placed elsewhere (another file) and we can source it in, so they are still visible to this file - we can pair on this if you like.

File: _utility/send_gh_comment.sh (hunk 3 above)
Comment (PaNaVTEC) [reply _utility/send_gh_comment.sh]: That is done in "send_gh_comment_validation.sh" but we splitted this in 2 scripts due to its verbosity 👍

File: _utility/send_gh_comment_validation.sh (hunk 7 above)
Comment (PaNaVTEC) [reply _utility/send_gh_comment_validation.sh:5]: This file has been autogenerated, I rather prefer leave it like it is, so if someone needs to add a new parameter in the future its as easy as autogenerate it in the webpage that I wrote in the comment in the header description

**Comment by PaNaVTEC** [comment]
@cyplo @neomatrix369 I addressed all your comments, if you are happy with the PR, :shipit:

File: _utility/send_gh_comment_validation.sh (hunk 7 above)
Comment (neomatrix369) [reply _utility/send_gh_comment_validation.sh:5]: I think you just did what I suggested so that cool.

**Comment by neomatrix369** [comment]
It looks good to me, @cyplo you have any comments?

**Pull request merged**